File: longParameterList.js
// 坏味道: 过长的参数列表 (超过5个参数)
function createUser(firstName, lastName, email, phone, address, city, state, zipCode, country, dateOfBirth, socialSecurityNumber, driverLicenseNumber, passportNumber, emergencyContactName, emergencyContactPhone, emergencyContactRelationship, medicalConditions, allergies, medications, insuranceProvider, insurancePolicyNumber, bankAccountNumber, routingNumber, creditCardNumber, creditCardExpiry, creditCardCVV) {
    console.log('Creating user with ' + firstName + ' ' + lastName);
    // 函数体...
}

function processOrder(orderId, customerId, productId, quantity, unitPrice, totalPrice, taxRate, taxAmount, discountRate, discountAmount, shippingAddress, billingAddress, paymentMethod, cardNumber, cardExpiry, cardCVV, shippingMethod, trackingNumber, orderDate, expectedDelivery, actualDelivery, status, notes, createdBy, updatedBy, createdAt, updatedAt) {
    console.log('Processing order ' + orderId);
    // 函数体...
}

function sendEmail(to, cc, bcc, subject, body, attachmentPath, attachmentName, attachmentType, priority, replyTo, from, signature, template, language, encoding, charset, format, timezone, scheduledTime, expirationTime, trackingId) {
    console.log('Sending email to ' + to);
    // 函数体...
}

function validateForm(username, password, confirmPassword, email, confirmEmail, firstName, middleName, lastName, dateOfBirth, gender, phone, mobile, address, city, state, zipCode, country, occupation, company, website, bio, interests, skills, education, experience, certifications, languages, timezone, locale, currency, termsAccepted, privacyAccepted, marketingAccepted) {
    console.log('Validating form for ' + username);
    // 函数体...
}

function createProduct(name, description, category, brand, price, cost, sku, barcode, weight, dimensions, color, size, material, manufacturer, supplier, warranty, returnPolicy, shippingClass, taxClass, status, isActive, isFeatured, isOnSale, salePrice, saleStartDate, saleEndDate, inventory, reorderPoint, maxStock, minStock, unit, packageSize, packageWeight, packageDimensions, countryOfOrigin, certifications, safetyWarnings, ingredients, allergens, nutritionalInfo, instructions, tags, metaTitle, metaDescription, metaKeywords, seoUrl, canonicalUrl, redirectUrl, customFields) {
    console.log('Creating product: ' + name);
    // 函数体...
}

module.exports = { 
    createUser, 
    processOrder, 
    sendEmail, 
    validateForm, 
    createProduct 
};

File: LongParameterList.java
public class LongParameterList {
    
    // 坏味道: 过长的参数列表 (超过5个参数)
    public void createUser(String firstName, String lastName, String email, String phone, String address, String city, String state, String zipCode, String country, String dateOfBirth, String socialSecurityNumber, String driverLicenseNumber, String passportNumber, String emergencyContactName, String emergencyContactPhone, String emergencyContactRelationship, String medicalConditions, String allergies, String medications, String insuranceProvider, String insurancePolicyNumber, String bankAccountNumber, String routingNumber, String creditCardNumber, String creditCardExpiry, String creditCardCVV) {
        System.out.println("Creating user with " + firstName + " " + lastName);
        // 方法体...
    }
    
    public void processOrder(String orderId, String customerId, String productId, String quantity, String unitPrice, String totalPrice, String taxRate, String taxAmount, String discountRate, String discountAmount, String shippingAddress, String billingAddress, String paymentMethod, String cardNumber, String cardExpiry, String cardCVV, String shippingMethod, String trackingNumber, String orderDate, String expectedDelivery, String actualDelivery, String status, String notes, String createdBy, String updatedBy, String createdAt, String updatedAt) {
        System.out.println("Processing order " + orderId);
        // 方法体...
    }
    
    public void sendEmail(String to, String cc, String bcc, String subject, String body, String attachmentPath, String attachmentName, String attachmentType, String priority, String replyTo, String from, String signature, String template, String language, String encoding, String charset, String format, String timezone, String scheduledTime, String expirationTime, String trackingId) {
        System.out.println("Sending email to " + to);
        // 方法体...
    }
    
    public void validateForm(String username, String password, String confirmPassword, String email, String confirmEmail, String firstName, String middleName, String lastName, String dateOfBirth, String gender, String phone, String mobile, String address, String city, String state, String zipCode, String country, String occupation, String company, String website, String bio, String interests, String skills, String education, String experience, String certifications, String languages, String timezone, String locale, String currency, String termsAccepted, String privacyAccepted, String marketingAccepted) {
        System.out.println("Validating form for " + username);
        // 方法体...
    }
}
